api/v1alpha1/jobset_types.go

```diff
@@ -110,12 +117,33 @@ type Network struct {
 	EnableDNSHostnames *bool `json:"enableDNSHostnames,omitempty"`
 }
 
+// Operator defines the target of a SuccessPolicy or FailurePolicy.
+type Operator string
+
+const (
+	// OperatorAll applies to all jobs matching the jobSelector.
+	OperatorAll Operator = "All"
+
+	// OperatorAny applies to any single job matching the jobSelector.
+	OperatorAny Operator = "Any"
+)
+
 type FailurePolicy struct {
 	// MaxRestarts defines the limit on the number of JobSet restarts.
 	// A restart is achieved by recreating all active child jobs.
 	MaxRestarts int `json:"maxRestarts,omitempty"`
 }
 
+type SuccessPolicy struct {
+	// Operator determines either All or Any of the selected jobs should succeed to consider the JobSet successful
+	// +kubebuilder:validation:Enum=All;Any
+	Operator Operator `json:"operator"`
+
+	// ReplicatedJobNames are the names of the replicated jobs the operator will apply to.
+	// An empty list will apply to all replicatedJobs.
```

Comment:
So a user would say replicatedJobNames: []string{} to mean apply to all?  Or we assume that if replicatedJobNames is nil then we apply for all if someone sets something in SuccessPolicy?

Reply:
NVM, I see below you answer this question.  I find this a bit confusing but maybe its a common pattern in Kubernetes?

Reply:
I am reading this as by default we will apply `All` and users don't need to specify anything.  But if they want something more complicated they have specify `Any` and give a list of names.

Reply:
> NVM, I see below you answer this question. I find this a bit confusing but maybe its a common pattern in Kubernetes?

I am drawing inspiration from the precedent of an empty label selector selecting all of the given type of object, for example. It also seems like a simpler user-experience especially for JobSets with a large number of replicatedJobs who simply wants to wait until all jobs are done to succeed the Jobset. 

Perhaps we should rename this field `SpecificReplicatedJobs` or something to more accurately convey the semantics of the field (i.e., leaving it empty selects no specific replicatedJobs and applies to them all by default)?

> I am reading this as by default we will apply All and users don't need to specify anything. But if they want something more complicated they have specify Any and give a list of names.

Correct.